terraform/cache/services/provider_cache.go

```diff
@@ -226,6 +243,26 @@ func (cache *ProviderCache) warmUp(ctx context.Context) error {
 	return nil
 }
 
+func (cache *ProviderCache) newRequest(ctx context.Context, url string) (*http.Request, error) {
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, url, nil)
+	if err != nil {
+		return nil, errors.WithStackTrace(err)
+	}
+
+	if cache.credsSource == nil {
+		return req, nil
+	}
+
+	hostname := svchost.Hostname(req.URL.Hostname())
+	if creds, err := cache.credsSource.ForHost(hostname); err != nil {
+		return nil, err
+	} else if creds != nil {
```

Comment:
I think `else` can be omitted here, since if condition have return

Reply:
In fact, the error returned is always `nil`, so I get rid of the error altogether (improved coed).